tools/run-e2e-test.sh

```diff
@@ -0,0 +1,177 @@
+VERBOSITY=2
+DEFAULT_GCP_ZONE="us-central1-b"
+# run-id is set to milliseconds since the unix epoch.
+RUN_ID="$(date +%s%N | cut -b1-13)"
+RUN_DIR="${REPO_ROOT}/_rundir/${RUN_ID}"
+# Ensure GOPATH/GOBIN are set.
+if [[ ! -v GOPATH ]]; then
+    GOPATH="${HOME}/go"
+fi
+if [[ ! -v GOBIN ]]; then
+    GOBIN="${GOPATH}/bin"
+fi
+
+# Trap interrupt, error, and exit to cleanup.
+cleanup() {
+    # The PROJECT env var is set when acquiring a boskos project.
+    if [[ -v PROJECT ]]; then
+	echo "Cleaning up boskos..."
+	cleanup_boskos
+    fi
+
+    echo "Cleaning up run dir: ${RUN_DIR}"
+    rm -rf ${RUN_DIR}
+}
+trap cleanup INT TERM EXIT
+
+# Set up (for cli runs) or acquire a (in case of prow CI) GCP project
+# for bringing up the cluster.
+bazel_user=""
+if [[ ! -v  GCP_PROJECT ]]; then
+    # GCP_PROJECT not defined, so attempt to acquire a boskos project
+    echo "No GCP Project specified...attempting to acquire boskos project"
+    acquire_project
+    GCP_PROJECT=${PROJECT}
+    echo "Boskos Project: ${GCP_PROJECT}"
+    # TODO: FIGURE OUT HOW TO GET ZONE BOSKOS PROJECT IS IN
+    if [[ ! -v GCP_ZONE ]]; then
+	GCP_ZONE=${DEFAULT_GCP_ZONE}
+    fi
+    echo "GCP ZONE: ${GCP_ZONE}"
+else
+    echo "GCP Project specified: ${GCP_PROJECT}"
+    if [[ ! -v GCP_ZONE ]]; then
+	GCP_ZONE=${DEFAULT_GCP_ZONE}
+    fi
+    echo "GCP ZONE: ${GCP_ZONE}"
+    # Install bazel in user directory if running from command-line.
+    bazel_user="--user"
+fi
+gcp_project_args="--gcp-project ${GCP_PROJECT} --gcp-zone ${GCP_ZONE} "
+test_args="--provider=gce --gce-project=${GCP_PROJECT} --gce-zone=${GCP_ZONE} --minStartupPods=8"
+
+# This script is currently using the "use-built-binaries" option, which
+# requires the built "e2e.test" binary as well as the "kubectl" and "ginkgo"
+# binaries to exist in the "_rundir/${RUN_ID}" directory.
+
+# Create the run directory if it does not exist.
+#
+if [ ! -d "${RUN_DIR}" ]; then
+    echo "Creating run dir: ${RUN_DIR}"
+    mkdir -p "${RUN_DIR}"
+fi
+
+echo "Downloading/installing kubetest2"
+go install sigs.k8s.io/kubetest2@latest
+
+echo "Downloading/installing kubetest2 gce deployer"
+go install sigs.k8s.io/kubetest2/kubetest2-gce@latest
+
+echo "Downloading/installing kubetest2 ginkgo tester"
+go install sigs.k8s.io/kubetest2/kubetest2-tester-ginkgo@latest
+
+echo "Downloading/installing ginkgo binary"
+go install github.com/onsi/ginkgo/v2/ginkgo@latest
+cp "${GOBIN}/ginkgo" "${RUN_DIR}/ginkgo"
+
+echo "Downloading installing kubectl (${VERSION})"
+curl -LO -s "https://dl.k8s.io/release/${VERSION}/bin/linux/amd64/kubectl"
+chmod u+x ./kubectl
+mv -f ./kubectl "${RUN_DIR}/kubectl"
+echo
+
+# Build the e2e.test binary and position it correctly.
+#
+echo "Build the e2e.test binary"
+pushd "${REPO_ROOT}/test/e2e"
+go test -c
+popd
+echo "Move e2e.test binary to: ${RUN_DIR}"
+mv -f "${REPO_ROOT}/test/e2e/e2e.test" "${RUN_DIR}/e2e.test"
+
+# Install bazel
+#
+BAZEL_VERSION="5.4.0"
```

Comment:
TODO: setup .bazel-version in this repo (and/or remove bazel) and respect that here (it's like GOTOOLCHAIN)

Reply:
Fixed.